### PR TITLE
Add rail from osm

### PR DESCRIPTION
**Pull request by egozverev**
**Summary:** 
I've ported the code related to osm rail from [datasources-etl](https://github.com/mapaction/datasources-etl). 
This pull request corresponds to the [jira issue](https://mapaction.atlassian.net/browse/DATAPIPE-53?jql=project%20%3D%20DATAPIPE%20AND%20labels%20%3D%20good-first-issue)
*Note:* this is a pull request created by a GSoC candidate (me). This code runs locally and creates all necessary airflow tasks. I didn't have an opportunity to verify how it behaves with real data. This pull request **must** be checked on GCP before being accepted. 

**Details:**
* ported osm rail schema from [datasources-etl](https://github.com/mapaction/datasources-etl)
* added osm rail transform operator
* updated config with osm rail info
* added airflow instances for osm rail (extract, transform)

### CODE REVIEW

File: dags/config/config.yaml [hunk 1]
```diff
@@ -102,3 +117,4 @@ cmf:
     top-level: 2_Active_Data
     admin: 202_adm
     roads: 232_tran
+    rail: 242_tran  # may not fit into naming convention!!!
```
Comment (andrewphilipsmith) [inline dags/config/config.yaml]: One tiny thing, the folder `232_tran` is for all transport data, so includes roads and rail. Hence the correct value here is:
`rail: 232_tran`

**Comment by andrewphilipsmith** [comment]
Hi Egor, Thank you for the pull request. I have tried this in my local development environment and it worked perfectly on the first attempt. I will try and run it on Google Cloud tomorrow and let you know how I get on. I have made one comment about the output folder name - but it is a tiny detail.

Excellent work! :tada:

**Comment by egozverev** [comment]
Hi Andy! Thank you for checking the pull request. 
I've fixed the wrong folder name and pushed the changes.

Hope it will be executed smoothly on GCP.

**Comment by andrewphilipsmith** [comment]
Hi Egor, I have tested this on GCP and it worked correctly.

The tasks for countries that do not have railways in the real world failed, however, I do not consider this a problem with the code for the reasons below:

Some countries do not have railways and hence there is no data in this case. A user would like to be able to look at the output data (without knowledge of airflow) and distinguish between:
* "We have looked and concluded there are no railways" 
* "We have not even looked for railway data yet"

We do not yet have an agreed means of representing these scenarios in the output data (we have started experimenting with some ideas, though these need to be documented). In the absence of an agreement, this PR works as expected.

I am very happy to merge this PR. Thank you for your work.

**Comment by egozverev** [comment]
Hi Andy!

That is so cool it worked out with GCP! I wasn't able to test it with GCP so I suspected there would be problems with these changes. I am glad these problems are related to the absence of the agreement rather than some bugs in the code. 

I am actually quite excited. Thank you for providing this problem to me. It was really appropriate for GSoC. The task wasn’t hard for anyone familiar with the codebase. Though it still had some challenges for a newcomer. Solving it allowed me to get more or less familiar with the pipeline code. Which is for sure a good thing if I am going to work with this code.

**Pull request merged**